src/github.com/couchbase/sync_gateway/base/bucket.go

```diff
@@ -325,7 +325,7 @@ func (bucket CouchbaseBucket) CBSVersion() (major uint64, minor uint64, micro st
 }
 
 // Creates a Bucket that talks to a real live Couchbase server.
-func GetCouchbaseBucket(spec BucketSpec) (bucket Bucket, err error) {
+func GetCouchbaseBucket(spec BucketSpec, callback func(bucket string, err error)) (bucket Bucket, err error) {
```

Comment:
Might be better to define the callback function signature, probably in sg-bucket.

It's already defined in go-couchbase (from streaming.go):
```
type NotifyFn func(bucket string, err error)
```
but I suppose we'd need to define a version that's compatible with walrus/gocb.